.ai/auto-review/github_reviewer.py

```diff
@@ -92,11 +93,31 @@ def main():
             for response in responses:
                 if response.line:
                     result = post_line_comment(github=github, file=file, text=response.text, line=response.line)
+                if not result:
```

Comment:
96 : Incorrect logic: `if not result` condition might lead to incorrect diff_hunks usage. `diff_hunks` is calculated and passed to `post_line_comment_with_diff_hunks` even if the previous `post_line_comment` was successful. : Remove the `if not result` condition before the `diff_hunks = extract_diff_hunks(file_diffs)` line and before the subsequent call to `post_line_comment_with_diff_hunks`. Instead, always calculate the diff hunks and pass them to the new function `post_line_comment_with_diff_hunks`, removing the first call to `post_line_comment`.